src/Components/Context/TaskProvider.tsx

```diff
@@ -0,0 +1,40 @@
+import { createContext, useState, useEffect, useContext } from "react";
+import type { Task } from "../../Types/type";
+
+interface TaskContextProps {
+  tasks: Task[];
+  setTasks: React.Dispatch<React.SetStateAction<Task[]>>;
+}
+
+export const TaskContext = createContext<TaskContextProps | undefined>(
+  undefined
+);
+
+export const useTasks = () => {
+  const context = useContext(TaskContext);
+  if (!context) {
+    throw new Error("useTasks must be used within a TaskProvider");
+  }
+  return context;
+};
+
+const saveTasksToLocalStorage = (tasks: Task[]) => {
+  localStorage.setItem("tasks", JSON.stringify(tasks));
+};
+
+export const TaskProvider: React.FC<{ children: React.ReactNode }> = ({
```

Comment:
interface TaskProviderProps {
  children: React.ReactNode;
}

```suggestion
export const TaskProvider: React.FC<TaskProviderProps> = ({
```